flocker/common/_retry.py

```diff
@@ -112,6 +112,32 @@ def loop(result):
     return d.addActionFinish()
 
 
+def timeout(reactor, deferred, timeout_sec):
+    """Adds a timeout to an existing deferred. If the timeout expires before
+    the deferred expires, then the deferred is cancelled.
+
+    :param reactor: The reactor implementation to schedule the timeout.
+    :type reactor: ``IReactorTime``.
+
+    :param deferred: The deferred to cancel at a later point in time.
+    :type deferred: ``Deferred``.
+
+    :param timeout_sec: The number of seconds to wait before the deferred
+                        should time out.
+    :type timeout_sec: ``float``.
```

Comment:
Nit: I think it's OK to incorporate the type into the param declaration, e.g. `param IReactorTime reactor:`